src/overlays/actors/ovl_En_Bat/z_en_bat.c

```diff
@@ -33,99 +40,522 @@ const ActorInit En_Bat_InitVars = {
+    this->actor.speedXZ *= Math_CosS(this->actor.world.rot.x);
+    this->actor.bgCheckFlags &= ~1;
+    this->actor.velocity.y = 0.0f;
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_FFLY_DEAD);
+
+    if (this->actor.colChkInfo.damageEffect == BAD_BAT_DMGEFF_ICE) {
+        this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX;
+        this->drawDmgEffAlpha = 1.0f;
+        this->drawDmgEffFrozenSteamScale = 60.0f * (0.45f / 40.0f);
+        this->drawDmgEffScale = 0.45f;
+    } else if (this->actor.colChkInfo.damageEffect == BAD_BAT_DMGEFF_LIGHT) {
+        this->drawDmgEffType = ACTOR_DRAW_DMGEFF_LIGHT_ORBS;
+        this->drawDmgEffAlpha = 4.0f;
+        this->drawDmgEffScale = 0.45f;
+        Actor_Spawn(&play->actorCtx, play, ACTOR_EN_CLEAR_TAG, this->collider.info.bumper.hitPos.x,
+                    this->collider.info.bumper.hitPos.y, this->collider.info.bumper.hitPos.z, 0, 0, 0,
+                    CLEAR_TAG_SMALL_LIGHT_RAYS);
+    } else if (this->actor.colChkInfo.damageEffect == BAD_BAT_DMGEFF_FIRE) {
+        this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FIRE;
+        this->drawDmgEffAlpha = 4.0f;
+        this->drawDmgEffScale = 0.45f;
+    }
+
+    Actor_SetColorFilter(&this->actor, 0x4000, 255, 0, 40);
+
+    if (this->actor.flags & ACTOR_FLAG_8000) {
+        this->actor.speedXZ = 0.0f;
+    }
+
+    this->collider.base.acFlags &= ~AC_ON;
+    this->actor.flags |= ACTOR_FLAG_10;
+    this->actionFunc = EnBat_Die;
+}
+
+void EnBat_Die(EnBat* this, PlayState* play) {
+    Math_StepToF(&this->actor.speedXZ, 0.0f, 0.5f);
+    this->actor.colorFilterTimer = 40;
+
+    if (!(this->actor.flags & ACTOR_FLAG_8000)) { // Carried by arrow
+        if (this->drawDmgEffType != ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX) {
+            Math_ScaledStepToS(&this->actor.shape.rot.x, 0x4000, 0x200);
+            this->actor.shape.rot.z += 0x1780;
+        }
+
+        if ((this->actor.bgCheckFlags & 1) || (this->actor.floorHeight == -32000.0f)) {
+            if (this->drawDmgEffType == ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX) {
+                Actor_SpawnIceEffects(play, &this->actor, this->bodyPartPoss, ARRAY_COUNT(this->bodyPartPoss), 2, 0.2f,
+                                      0.2f);
+            }
+
+            func_800B3030(play, &this->actor.world.pos, &gZeroVec3f, &gZeroVec3f, 100, 0, 0);
+            SoundSource_PlaySfxAtFixedWorldPos(play, &this->actor.world.pos, 11, NA_SE_EN_EXTINCT);
+            Actor_MarkForDeath(&this->actor);
+
+            if (this->actor.room == -1) {
+                Actor* actor = NULL;
+
+                // Search for other EnBats. If find none, set switch flag.
+                do {
+                    actor = SubS_FindActor(play, actor, ACTORCAT_ENEMY, ACTOR_EN_BAT);
+                    if (actor != NULL) {
+                        if (actor == &this->actor) {
+                            actor = actor->next;
+                        } else {
+                            break;
+                        }
```

Comment:
```suggestion
                        if (actor == &this->actor) {
                            actor = actor->next;
                            continue;
                        }
                        break;
```
I don't know if this is better, but it likely matches. So I'll give you the option

Reply:
Strangely this was one I hadn't tried, I think it probably is a bit better.

Reply:
Actually, this works too:
```c
                do {
                    enemy = SubS_FindActor(play, enemy, ACTORCAT_ENEMY, ACTOR_EN_BAT);
                    if (enemy != NULL) {
                        if (enemy != &this->actor) {
                            break;
                        }
                        enemy = enemy->next;
                    }
                } while (enemy != NULL);
```